tests/parametric/test_telemetry.py

```diff
@@ -0,0 +1,189 @@
+        ), "Install time should be the propagated value, got {}".format(install_time)
+
+        assert (
+            "_dd.install.type" in traces[0][0]["meta"]
+        ), "The installation type should included in the first span of the first trace, got {}".format(
+            traces[0][0]["meta"]
+        )
+        install_type = traces[0][0]["meta"]["_dd.install.type"]
+        assert install_type == "k8s_single_step", "Install type should be the propagated value"
+
+        assert (
+            "_dd.install.id" in traces[0][0]["meta"]
+        ), "The installation id should included in the first span of the first trace, got {}".format(
+            traces[0][0]["meta"]
+        )
+        install_id = traces[0][0]["meta"]["_dd.install.id"]
+        assert (
+            install_id == library_env["DD_INSTRUMENTATION_INSTALL_ID"]
+        ), "Install id should be the propagated value, got {}".format(install_id)
+
+        # Ensure other spans do not have the telemetry
+        for span in (traces[0][1], traces[1][0]):
+            assert "_dd.install.time" not in span["meta"]
+            assert "_dd.install.type" not in span["meta"]
+            assert "_dd.install.id" not in span["meta"]
+
+    @pytest.mark.parametrize("library_env", [{**DEFAULT_ENVVARS}])
+    def test_first_trace_telemetry_not_propagated(self, library_env, test_agent, test_library):
+        """
+        When instrumentation data is not propagated to the library
+            The first trace should not contain telemetry as the Agent will add it when not present.
+        """
+        with test_library.start_span("first_span"):
+            pass
+
+        traces = test_agent.wait_for_num_traces(num=1)
+
+        assert (
+            "_dd.install.time" not in traces[0][0]["meta"]
+        ), "The installation time should not be included in the first span of the first trace, got {}".format(
+            traces[0][0]["meta"]["_dd.install.time"]
+        )
+        assert (
+            "_dd.install.type" not in traces[0][0]["meta"]
+        ), "The installation type should not be included in the first span of the first trace, got {}".format(
+            traces[0][0]["meta"]["_dd.install.type"]
+        )
+        assert (
+            "_dd.install.id" not in traces[0][0]["meta"]
+        ), "The installation id should not be included in the first span of the first trace, got {}".format(
+            traces[0][0]["meta"]
+        )
+
+    @pytest.mark.parametrize(
+        "library_env",
+        [
+            {
+                **DEFAULT_ENVVARS,
+                "DD_INSTRUMENTATION_INSTALL_TIME": str(int(time.time())),
+                "DD_INSTRUMENTATION_INSTALL_TYPE": "k8s_single_step",
+                "DD_INSTRUMENTATION_INSTALL_ID": str(uuid.uuid4()),
+            },
+        ],
+    )
+    def test_telemetry_event_propagated(self, library_env, test_agent, test_library):
+        """Ensure the installation ID is included in the app-started telemetry event.
```

Comment:
Does this only need to be in `app-started`? In the Python implementation we add these headers onto every request.